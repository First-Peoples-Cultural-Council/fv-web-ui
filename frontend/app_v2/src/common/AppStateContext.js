import React from 'react'
const AppStateContext = React.createContext({
  id: 'AppStateContext',
})

export default AppStateContext
