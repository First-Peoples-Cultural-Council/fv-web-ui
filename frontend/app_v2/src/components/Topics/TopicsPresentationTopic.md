```jsx
    <TopicsPresentationTopic
    audio='dist/styleguide/TestWav.wav'
    heading='W.1'
    image='dist/styleguide/TestPhoto.jpg'
    subheading='Everything'
    type='WIDGET_LIST_WORD'
    url='/word/everything'
    />
    <br />
    <TopicsPresentationTopic
    audio='dist/styleguide/TestWav.wav'
    heading='W.2'
    subheading='No img'
    type='WIDGET_LIST_WORD'
    url='/word/noImage'
    />
    <br />
    <TopicsPresentationTopic
    heading='W.3'
    subheading='No img, audio'
    type='WIDGET_LIST_WORD'
    url='/word/nothing'
    />
    <br />
    <TopicsPresentationTopic
    heading='St.1'
    image='dist/styleguide/TestPhoto.jpg'
    subheading='Everything'
    type='WIDGET_LIST_STORY'
    url='/story/everything'
    />
    <br />
    <TopicsPresentationTopic
    listCount='21'
    heading='St.2'
    subheading='No img'
    type='WIDGET_LIST_STORY'
    url='/story/noImage'
    />
    <br />
    <TopicsPresentationTopic
    listCount='20'
    heading='P.1'
    image='dist/styleguide/TestPhoto.jpg'
    subheading='Everything'
    type='WIDGET_LIST_PHRASE'
    url='/phrase/everything'
    />
    <br />
    <TopicsPresentationTopic
    listCount='21'
    heading='P.2'
    subheading='No img'
    type='WIDGET_LIST_PHRASE'
    url='/phrase/noImage'
    />
    <br />
    <TopicsPresentationTopic
    audio='dist/styleguide/TestWav.wav'
    heading='So.1'
    image='dist/styleguide/TestPhoto.jpg'
    subheading='Everything'
    type='WIDGET_LIST_SONG'
    url='/song/everything'
    />
    <br />
    <TopicsPresentationTopic
    audio='dist/styleguide/TestWav.wav'
    heading='So.2'
    subheading='No img'
    type='WIDGET_LIST_SONG'
    url='/song/noImage'
    />
    <br />
    <TopicsPresentationTopic
    heading='So.3'
    subheading='No img, audio'
    type='WIDGET_LIST_SONG'
    url='/song/nothing'
    />
```
