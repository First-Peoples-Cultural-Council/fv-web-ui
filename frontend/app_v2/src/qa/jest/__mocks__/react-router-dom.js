module.exports = {
  ...jest.requireActual('react-router-dom'),
  useLocation: ()=>{
    return {
      pathname: '/src/qa/jest/__mocks__/react-router-dom',
      search: '?page=1&pageSize=10&searchTerm=Ha%C3%AD%C9%ABzaqv%E1%B8%B7a&letter=%C9%AB&isTrue=true&isFalse=false',
      hash: '',
    }
  },
  useParams: ()=>{
    return {
      language: "jestLanguage",
      character: "k'",
    }
  },
}
