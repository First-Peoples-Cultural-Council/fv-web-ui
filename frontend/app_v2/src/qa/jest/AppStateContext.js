import React from 'react'
import AppStateContext from 'common/AppStateContext'
module.exports = {
  Provider: ({ children }) => {
    return (
      <AppStateContext.Provider
        value={{
          reducer: {
            state: undefined,
            dispatch: undefined,
          },
          audio: {
            machine: {
              value: undefined,
              context: {
                player: new Audio(),
                src: undefined,
                errored: [],
                scrubMs: 500,
              },
            },
            send: undefined,
          },
        }}
      >
        {children}
      </AppStateContext.Provider>
    )
  },
}
