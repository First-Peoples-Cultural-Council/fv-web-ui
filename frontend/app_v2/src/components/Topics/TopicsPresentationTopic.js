import React from 'react'
import PropTypes from 'prop-types'
import useIcon from 'common/useIcon'
import AudioMinimal from 'components/AudioMinimal'
import { Link } from 'react-router-dom'
import { useParams } from 'react-router-dom'
import {
  WIDGET_LIST_WORD,
  WIDGET_LIST_PHRASE,
  WIDGET_LIST_SONG,
  WIDGET_LIST_STORY,
  // WIDGET_LIST_MIXED,
  // WIDGET_LIST_GENERIC,
} from 'common/constants'
/**
 * @summary TopicsPresentationTopic
 * @component
 *
 * @param {object} props
 *
 * @returns {node} jsx markup
 */
function TopicsPresentationTopic({ audio, heading, image, listCount, subheading, type, url: _url }) {
  const { sitename } = useParams()
  const url = `${sitename}/${_url}`
  let bgColor
  let icon
  let textSize
  switch (type) {
    case WIDGET_LIST_WORD:
      bgColor = 'bg-fv-turquoise'
      icon = 'ChatBubble'
      textSize = 'text-4xl lg:text-5xl'
      break
    case WIDGET_LIST_PHRASE:
      bgColor = 'bg-fv-orange'
      icon = 'Quote'
      textSize = 'text-3xl lg:text-4xl'
      break
    case WIDGET_LIST_SONG:
      bgColor = 'bg-fv-red'
      icon = 'MusicNote'
      textSize = 'text-3xl lg:text-4xl'
      break
    case WIDGET_LIST_STORY:
      bgColor = 'bg-fv-purple'
      icon = 'Book'
      textSize = 'text-3xl lg:text-4xl'
      break

    default:
      // nothing
      break
  }
  return (
    <div
      data-testid="TopicsPresentationTopic"
      style={
        image
          ? {
              backgroundImage: `linear-gradient(to bottom, rgba(0, 0, 0, 0.4), rgba(0, 0, 0, 0.3)), url(${image})`,
            }
          : {}
      }
      className={`
        ${bgColor}
        bg-center
        bg-cover
        flex
        flex-col
        items-center
        justify-between
        rounded-25
        xl:rounded-50
        p-8
        text-white
        w-full
      `}
    >
      {useIcon(icon, `${image ? 'fill-current' : 'opacity-30'} w-7 h-7 lg:w-10 lg:h-10 xl:w-14 xl:h-14`)}
      {heading && (
        <h1 className={`${textSize} text-center font-medium my-3`}>
          <Link to={url}>{heading}</Link>
        </h1>
      )}
      {subheading && <h2 className="text-xl lg:text-2xl text-center">{subheading}</h2>}
      {audio && <AudioMinimal.Container src={audio} iconStyling="fill-current w-7 h-7 lg:w-8 lg:h-8 mt-3" />}

      {listCount > 0 && (
        <div className="text-lg lg:text-xl text-center text-fv-yellow">
          {listCount === 1 && `${listCount} phrase`}
          {listCount > 1 && `${listCount} phrases`}
        </div>
      )}
    </div>
  )
}
// PROPTYPES
const { string, number } = PropTypes
TopicsPresentationTopic.propTypes = {
  audio: string,
  heading: string,
  image: string,
  listCount: number,
  subheading: string,
  type: string,
  url: string,
}
export default TopicsPresentationTopic
