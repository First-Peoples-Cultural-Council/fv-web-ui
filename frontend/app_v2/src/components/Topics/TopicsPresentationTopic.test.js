import React from 'react'
import { render, screen } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import TopicsPresentationTopic from 'components/Topics/TopicsPresentationTopic'
import AppStateContext from 'qa/jest/AppStateContext'

import {
  WIDGET_LIST_WORD,
  WIDGET_LIST_PHRASE,
  //   WIDGET_LIST_SONG,
  WIDGET_LIST_STORY,
  // WIDGET_LIST_MIXED,
  // WIDGET_LIST_GENERIC,
} from 'common/constants'
// Presentation
describe('TopicsPresentation', () => {
  const heading = 'lorem'
  const suheading = 'ipsum'
  const image = 'imageContet.jpg'
  const audio = 'audio'
  const url = 'url'
  const count = 99

  // WIDGET_LIST_WORD
  // ------------------------------------------
  test('WIDGET_LIST_WORD', () => {
    render(
      <Router>
        <AppStateContext.Provider>
          <TopicsPresentationTopic
            audio={audio}
            heading={heading}
            image={image}
            subheading={suheading}
            url={url}
            type={WIDGET_LIST_WORD}
          />
        </AppStateContext.Provider>
      </Router>
    )

    // Heading
    expect(
      screen.getByText(heading, {
        exact: false,
      })
    ).toBeInTheDocument()

    // Subheading
    expect(
      screen.getByText(suheading, {
        exact: false,
      })
    ).toBeInTheDocument()

    // Icon
    expect(screen.getByTestId('ChatBubble')).toBeInTheDocument()

    // Audio
    expect(
      screen.getByText('Play audio', {
        exact: false,
      })
    ).toBeInTheDocument()

    expect(screen.getByTestId('TopicsPresentationTopic')).toMatchSnapshot()
  })

  // WIDGET_LIST_PHRASE
  // ------------------------------------------
  test('WIDGET_LIST_PHRASE', () => {
    render(
      <Router>
        <AppStateContext.Provider>
          <TopicsPresentationTopic
            heading={heading}
            image={image}
            url={url}
            listCount={count}
            type={WIDGET_LIST_PHRASE}
          />
        </AppStateContext.Provider>
      </Router>
    )

    // Heading
    expect(
      screen.getByText(heading, {
        exact: false,
      })
    ).toBeInTheDocument()

    // Icon
    expect(screen.getByTestId('Quote')).toBeInTheDocument()

    // Count
    expect(
      screen.getByText('99 phrases', {
        exact: false,
      })
    ).toBeInTheDocument()

    expect(screen.getByTestId('TopicsPresentationTopic')).toMatchSnapshot()
  })

  // WIDGET_LIST_STORY
  // ------------------------------------------
  test('WIDGET_LIST_STORY', () => {
    render(
      <Router>
        <AppStateContext.Provider>
          <TopicsPresentationTopic
            heading={heading}
            subheading={suheading}
            image={image}
            url={url}
            type={WIDGET_LIST_STORY}
          />
        </AppStateContext.Provider>
      </Router>
    )

    // Heading
    expect(
      screen.getByText(heading, {
        exact: false,
      })
    ).toBeInTheDocument()

    // Subheading
    expect(
      screen.getByText(suheading, {
        exact: false,
      })
    ).toBeInTheDocument()

    // Icon
    expect(screen.getByTestId('Book')).toBeInTheDocument()

    expect(screen.getByTestId('TopicsPresentationTopic')).toMatchSnapshot()
  })
})
