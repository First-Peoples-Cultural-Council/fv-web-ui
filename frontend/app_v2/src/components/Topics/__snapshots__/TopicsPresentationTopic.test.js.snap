// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicsPresentation WIDGET_LIST_PHRASE 1`] = `
<div
  class="
        bg-fv-orange
        bg-center
        bg-cover
        flex
        flex-col
        items-center
        justify-between
        rounded-25
        xl:rounded-50
        py-6
        text-white
        w-full
      "
  data-testid="TopicsPresentationTopic"
>
  <svg
    class="
        fill-current
        w-7
        h-7
        lg:w-8
        lg:h-8
        xl:w-12
        xl:h-12
      "
    data-testid="Quote"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M6 17h3l2-4V7H5v6h3zm8 0h3l2-4V7h-6v6h3z"
    />
  </svg>
  <h1
    class="
          text-xl
          text-center
          lg:text-2xl
          xl:text-5xl
          font-medium
        "
  >
    lorem
  </h1>
  <div>
    99 phrases
  </div>
</div>
`;

exports[`TopicsPresentation WIDGET_LIST_STORY 1`] = `
<div
  class="
        bg-fv-purple
        bg-center
        bg-cover
        flex
        flex-col
        items-center
        justify-between
        rounded-25
        xl:rounded-50
        py-6
        text-white
        w-full
      "
  data-testid="TopicsPresentationTopic"
>
  <svg
    class="
        fill-current
        w-7
        h-7
        lg:w-8
        lg:h-8
        xl:w-12
        xl:h-12
      "
    data-testid="Book"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12.52,5.59c0,0,4.44-2.95,9.44-2.95v16.69c0,0-6.75-0.56-9.44,2.13V5.59z"
    />
    <path
      d="M11.49,5.59c0,0-4.44-2.95-9.44-2.95v16.69c0,0,6.75-0.56,9.44,2.13V5.59z"
    />
  </svg>
  <h1
    class="
          text-xl
          text-center
          lg:text-2xl
          xl:text-5xl
          font-medium
        "
  >
    lorem
  </h1>
  <h2
    class="
          text-base
          lg:text-xl
          xl:text-3xl
        "
  >
    ipsum
  </h2>
</div>
`;

exports[`TopicsPresentation WIDGET_LIST_WORD 1`] = `
<div
  class="
        bg-fv-turquoise
        bg-center
        bg-cover
        flex
        flex-col
        items-center
        justify-between
        rounded-25
        xl:rounded-50
        py-6
        text-white
        w-full
      "
  data-testid="TopicsPresentationTopic"
>
  <svg
    class="
        fill-current
        w-7
        h-7
        lg:w-8
        lg:h-8
        xl:w-12
        xl:h-12
      "
    data-testid="ChatBubble"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
    <path
      d="M20 2H4c-1.1 0-2 .9-2 2v18l4-4h14c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2z"
    />
  </svg>
  <h1
    class="
          text-xl
          text-center
          lg:text-2xl
          xl:text-5xl
          font-medium
        "
  >
    lorem
  </h1>
  <h2
    class="
          text-base
          lg:text-xl
          xl:text-3xl
        "
  >
    ipsum
  </h2>
  <button
    aria-live="off"
    type="button"
  >
    <div
      class="sr-only"
    >
      Play audio
    </div>
    <svg
      class="
        fill-current
        w-7
        h-7
        lg:w-8
        lg:h-8
        xl:w-12
        xl:h-12
      "
      height="24"
      viewBox="0 0 512 512"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm115.7 272l-176 101c-15.8 8.8-35.7-2.5-35.7-21V152c0-18.4 19.8-29.8 35.7-21l176 107c16.4 9.2 16.4 32.9 0 42z"
      />
    </svg>
  </button>
</div>
`;
